### PR TITLE
Add tests for close only. Everything except increasing positions should work

### PR DESCRIPTION
**Pull request by bachstatter**
Tests for:
- flag
- liquidate
- increase/decrease pos
- deposit, withdraw and withdrawAll

### CODE REVIEW

File: test/integration/modules/LiquidationModule.test.ts [hunk 1]
```diff
@@ -435,7 +464,7 @@ describe('LiquidationModule', () => {
   });
 
   describe('liquidatePosition', () => {
-    const commtAndSettleLiquidatedPosition = async (desiredKeeper: ethers.Signer) => {
+    const commitAndSettleLiquidatedPosition = async (desiredKeeper: ethers.Signer) => {
```
Comment (davidvuong) [inline test/integration/modules/LiquidationModule.test.ts:467]: 😱

File: test/integration/modules/LiquidationModule.test.ts [hunk 2]
```diff
@@ -100,6 +100,35 @@ describe('LiquidationModule', () => {
       );
     });
 
+    it('should be able to flag when market is in close only', async () => {
```
Comment (davidvuong) [inline test/integration/modules/LiquidationModule.test.ts:103]: It's late for me but I don't see anywhere where `maxMarketSize=0` to make the market close-only 🤔

File: test/integration/modules/LiquidationModule.test.ts [hunk 3]
```diff
@@ -547,6 +576,40 @@ describe('LiquidationModule', () => {
...
+        .mul(orderSide === 1 ? 0.9 : 1.1)
+        .toBN();
+      await market.aggregator().mockSetCurrentPrice(newMarketOraclePrice);
+
+      await PerpMarketProxy.connect(keeper()).flagPosition(trader.accountId, marketId);
+
+      // Set market to close only
+      await setMarketConfigurationById(bs, marketId, { maxMarketSize: 0 });
+
+      const tx = await PerpMarketProxy.connect(keeper()).liquidatePosition(trader.accountId, marketId);
```
Comment (davidvuong) [inline test/integration/modules/LiquidationModule.test.ts:608]: `withExplicitEvmMine` here

File: test/integration/modules/MarginModule.test.ts [hunk 4]
```diff
@@ -241,7 +242,23 @@ describe('MarginModule', async () => {
...
-      it('should emit depositMarketUsd when using sUSD as collateral'); // To write this test we need to configure `globalConfig.usdToken` as collateral in bootstrap.
+      it('should allow deposit of collateral when collateral maxAllowable is 0', async () => {
+        const { PerpMarketProxy } = systems();
+
+        const { collateral, collateralDepositAmount, trader, market } = await depositMargin(bs, genTrader(bs));
+
+        await setMarketConfigurationById(bs, market.marketId(), { maxMarketSize: bn(0) });
+        await mintAndApprove(bs, collateral, collateralDepositAmount, trader.signer);
+        // Should also be able to deposit
+        const depositTx = await PerpMarketProxy.connect(trader.signer).modifyCollateral(
```
Comment (davidvuong) [inline test/integration/modules/MarginModule.test.ts:253]: `withExplicitEvmMine` here and below in `should allow withdraw when market is in close only`.

File: test/integration/modules/LiquidationModule.test.ts [hunk 5]
```diff
@@ -547,6 +576,40 @@ describe('LiquidationModule', () => {
...
+
+      const newMarketOraclePrice = wei(order.oraclePrice)
+        .mul(orderSide === 1 ? 0.9 : 1.1)
+        .toBN();
+      await market.aggregator().mockSetCurrentPrice(newMarketOraclePrice);
+
+      await PerpMarketProxy.connect(keeper()).flagPosition(trader.accountId, marketId);
+
+      // Set market to close only
+      await setMarketConfigurationById(bs, marketId, { maxMarketSize: 0 });
```
Comment (davidvuong) [inline test/integration/modules/LiquidationModule.test.ts:606]: Minor: Good to be extra safe here and `assertBn.isZero(await PerpMarketProxy.getMarketConfigurationById(marketId).maxMarketSize)`

File: test/integration/modules/OrderModule.test.ts [hunk 6]
```diff
@@ -173,6 +174,37 @@ describe('OrderModule', () => {
       );
     });
 
+    it('should be able to set maxMarketSize (oi) to 0 with open positions', async () => {
+      const { PerpMarketProxy } = systems();
+
+      const { trader, market, marketId, collateral, collateralDepositAmount } = await depositMargin(bs, genTrader(bs));
+      const order = await genOrder(bs, market, collateral, collateralDepositAmount);
+      await commitAndSettle(bs, marketId, trader, order);
+      // Update the market's maxMarketSize to be just slightly below (95%) sizeDelta. We .abs because order can be short.
```
Comment (davidvuong) [inline test/integration/modules/OrderModule.test.ts:183]: dis comment

File: test/integration/modules/MarginModule.test.ts [hunk 7]
```diff
@@ -589,6 +606,20 @@ describe('MarginModule', async () => {
         await assertEvent(receipt, `MarginWithdraw(${marginWithdrawEventProperties})`, PerpMarketProxy);
       });
 
+      it('should allow withdraw when market is in close only', async () => {
```
Comment (davidvuong) [inline test/integration/modules/MarginModule.test.ts:609]: I don't see a `maxMarketSize: 0` here either 🤔

Comment (bachstatter) [reply test/integration/modules/MarginModule.test.ts:616]: Wow sorry about that, moved a little too fast..

**Comment by bachstatter** [comment]
All addressed thanks for catching my sloppiness in this one!

**Comment by davidvuong** [comment]
LGTM merge it 👍

**Pull request merged**